primitives/runtime/src/lib.rs

```diff
@@ -782,6 +782,9 @@ pub type ApplyExtrinsicResult =
 pub type ApplyExtrinsicResultWithInfo<T> =
 	Result<DispatchResultWithInfo<T>, transaction_validity::TransactionValidityError>;
 
+/// The error type used as return type in try runtime hooks.
+pub type TryRuntimeError = DispatchError;
```

Comment:
I dont get the advantage of it either. cc @KiChjang   
We (ab)use `DispatchResult` everywhere, so just re-use it for try-runtime checks.

Reply:
I assume you are referring to `DispatchError` here since that is what you are pointing to. I think the benefit of this is better readability. But the question is whether it is worth it since this way we are creating an abstraction. 

Also referring to @kianenigma 's comment in the initial issue:
> This is slightly bad in terms of naming, as try-runtime code paths are not necessarily Dispatched, but we really want to reuse DispatchResult's ability to reference exactly which module it is coming from.

Reply:
That's what I understood from the issue as well. Essentially, a `TryRuntimeError` is a `DispatchError`, but it just doesn't feel correct to return a `DispatchError` from try-runtime as it is not executing a dispatch, hence we have this type alias.